File: CS/Demos/02ClassesAndModules/Runner.cs
namespace Demos.ClassesAndModules;

static class Runner
{
  public static void RunMe()
  {
    AddAndPrintValueWithModule();

    //var threads = new Thread[10];

    //for (var i = 0; i < threads.Length; i++)
    //{
    //  threads[i] = new Thread(AddAndPrintValueWithModule);
    //}

    //for (var i = 0; i < threads.Length; i++)
    //{
    //  threads[i].Start();
    //}
  }

  private static void AddAndPrintValueWithModule()
  {
    StaticVar.PrintCalls();
    StaticVar.PrintCalls();
    StaticVar.PrintCalls();
  }
}

File: VB/Demos/02ClassesAndModules/Runner.vb
Imports System.Threading

Namespace ClassesAndModules
  Module Runner
    Public Sub RunMe()
      AddAndPrintValueWithModule()

      'Dim threads(10) As Thread

      'For i As Integer = 0 To threads.Length - 1
      '  threads(i) = New Thread(AddressOf AddAndPrintValueWithModule)
      'Next

      'For i As Integer = 0 To threads.Length - 1
      '  threads(i).Start()
      'Next

    End Sub

    Private Sub AddAndPrintValueWithModule()
      PrintCalls()
      PrintCalls()
      PrintCalls()
    End Sub
  End Module
End Namespace
